src/utils/theme_utils.ts

```diff
@@ -95,6 +95,24 @@ export const blendColors = (background: string, foreground: string, opacity: num
         opacity,
     );
 
+    if (hex) {
+        let r = red.toString(16);
+        let g = green.toString(16);
+        let b = blue.toString(16);
+
+        if (r.length === 1) {
+            r = '0' + r;
```

Comment:
This makes me wish we had some sort of `sprintf` equivalent in JS. If this was Go, we could do this in a one-liner like `fmt.Sprintf("#%02X%02X%02X", red, green, blue)`.

Now that I've written that out, it also looks kind of like gibberish though